auditlog-fhir-server/src/main/java/ca/uhn/fhir/jpa/starter/MyConsentService.java

```diff
@@ -155,13 +166,43 @@ private void createAuditEvent(RequestDetails theRequestDetails, boolean wasSucce
     myAuditEventDao.create(auditEvent);
   }
 
-  //Array toString with removed [brackets] so the result looks like: value1,value2,value3
-  private String toCommaSeparatedStringList(String[] theInputStringArray) {
-    if (theInputStringArray.length > 0) {
-      String commaSeparatedListWithBrackets = Arrays.toString(theInputStringArray);
-      return commaSeparatedListWithBrackets.substring(1, commaSeparatedListWithBrackets.length() - 1);
+  //retrieve the first subject reference id (patient id) of a resource (not exhaustive, only for resources of the tested radiological workflow)
+  private String retrieveSubjectIdOrPatientId(RequestDetails theRequestDetails) {
+    //TODO: figure out how to transer caseId for the operation  http://endpoint/DiagnosticReport/$DiagnosticReportId/$fhirToCDA
+
+    // GET request
+    if (theRequestDetails.getRequestType().equals("GET")) {
+      //GET request that contains a subject parameter
+      if (theRequestDetails.getParameters() != null &&
+        !theRequestDetails.getParameters().isEmpty() &&
+        theRequestDetails.getParameters().get("subject") != null) {
+        return theRequestDetails.getParameters().get("subject")[0];
+        //GET request that contains a patient's id
+      } else if (theRequestDetails.getResourceName().equals("Patient") &&
+        theRequestDetails.getId() != null &&
+        theRequestDetails.getId().getValue() != null &&
+        !theRequestDetails.getId().getValue().isEmpty()) {
+        return theRequestDetails.getId().getValue();
+      }
+    }
+
+    //PUT, POST, DELETE... or even GET request as long as it does not contain a parameter "subject"
+    if (theRequestDetails.getResourceName().equals("Patient")) {
+      return theRequestDetails.getId().getValue();
```

Comment:
Where is the exact difference to 181-185? Your comment says, that this can also be executed on GET requests. 

Because when it is a get requests and there is no subject the if in line 181 is triggered. If the condition is false e.g. because getId is null. Then the statement in line 191 would produce a NullPointerException. Or am I overseeing sometihng.

Reply:
you are absolutely right, the upper part is redundant and the lower part is not checkt for nullpointerexeptions properly. will change that.